src/lib/dialog/dialog.ts

```diff
@@ -122,6 +124,20 @@ export class MdDialog {
 
     return state;
   }
+
+  /**
+   * Applies default options to the dialog config.
+   * @param {MdDialogConfig} config Config to be modified.
+   * @returns The new configuration object.
```

Comment:
Good point. Removed the types and renamed the argument to match the other methods.